lib/util/PromisifierUtil.js

```diff
@@ -0,0 +1,66 @@
+import { isFunction, some } from 'min-dash';
+
+export function promisifyIfNecessary(api) {
+
+  return function() {
+
+    // @oguz:
+    // The index of callback function depends on the presence of
+    // optional parameters; however, the APIs we are promisifying are
+    // requiring only one argument of Function type. Therefore we
+    // can detect if a callback is passed like this:
+    var hasCallback = some(arguments, isFunction);
+
+    if (hasCallback) {
+      console.warn(
+        'Warning: passing callbacks to ' + api.name +
+        ' API will be deprecated in the next major bpmn-js release.' +
+        ' Consider switching to promises instead. See the API documentation.'
+      );
+
+      return api.apply(this, arguments);
+    }
+
+    if (!window.Promise) {
+      throw new Error('Promises are not supported for this browser. Consider polyfilling Promises.');
+    }
+
+    // eslint-disable-next-line no-undef
```

Comment:
Declare a global `Promise` variable in eslint to get rid of the warning instead.